include/deal.II/fe/mapping_fe_field.h

```diff
@@ -62,50 +62,91 @@ DEAL_II_NAMESPACE_OPEN
  *    dhq.distribute_dofs(fesystem);
  *    Vector<double> eulerq(dhq.n_dofs());
  *    const ComponentMask mask(spacedim, true);
- *    MappingFE<dim,spacedim> map(eulerq, dhq, mask);
+ *    MappingFEField<dim,spacedim> map(eulerq, dhq, mask);
  *    map.update_euler_vector_using_triangulation(eulerq);
  * @endcode
-
-
  *
  * @author Luca Heltai, Marco Tezzele 2013, 2015
  */
 template <int dim, int spacedim=dim,
-          class DH=DoFHandler<dim,spacedim>,
-          class VECTOR=Vector<double> >
-class MappingFE : public Mapping<dim,spacedim>
+          class VECTOR=Vector<double>,
+          class DH=DoFHandler<dim,spacedim> >
+class MappingFEField : public Mapping<dim,spacedim>
 {
 public:
   /**
    * Constructor. The first argument is a VECTOR that specifies the
    * transformation of the domain from the reference to the current
-   * configuration. This is filled calling the method
-   * update_euler_vector_using_triangulation.
+   * configuration.
+   *
+   * In general this class decouples geometry from topology, allowing
+   * users to define geometries which are only topologically
+   * equivalent to the underlying Triangulation, but which may
+   * otherwise be arbitrary. Differently from what happens in
+   * MappingQEulerian, the FiniteElement field which is passed to the
+   * constructor is interpreted as an absolute geometrical
+   * configuration, therefore one has to make sure that the
+   * euler_vector actually represents a valid geometry (i.e., one with
+   * no inverted cells, or with no zero-volume cells).
+   *
+   * In order to facilitate the user, we provide an
+   * update_euler_vector_using_triangulation() method to initialize a
+   * euler_vector in a way similar to what happens in
+   * MappingQEulerian, by creating a euler_vector which interpolates
+   * the underlying Triangulation.
+   *
+   * If the underlying FiniteElement is a system of FE_Q(), and
+   * euler_vector is initialized using
+   * update_euler_vector_using_triangulation(), then this class is in
+   * all respects identical to MappingQ().
```

Comment:
This is identical to `MappingQEulerian`, with the exception that euler vector stores **absolute** positions instead of displacements. 

The class stores a reference to euler_vector, and does not own it. Just as in MappingQEulerian.